src/main/java/org/eclipse/tractusx/managedidentitywallets/KeyStorageService.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * *******************************************************************************
+ *  Copyright (c) 2021,2024 Contributors to the Eclipse Foundation
+ *
+ *  See the NOTICE file(s) distributed with this work for additional
+ *  information regarding copyright ownership.
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations
+ *  under the License.
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ * ******************************************************************************
+ */
+
+package org.eclipse.tractusx.managedidentitywallets;
+
+import org.eclipse.tractusx.managedidentitywallets.dao.entity.HoldersCredential;
+import org.eclipse.tractusx.managedidentitywallets.domain.HoldersCredentialCreationConfig;
+import org.eclipse.tractusx.managedidentitywallets.domain.KeyStorageType;
+import org.eclipse.tractusx.managedidentitywallets.domain.PresentationCreationConfig;
+import org.eclipse.tractusx.ssi.lib.crypt.KeyPair;
+import org.eclipse.tractusx.ssi.lib.exception.KeyGenerationException;
+
+public interface KeyStorageService {
+    HoldersCredential createHoldersCredential(HoldersCredentialCreationConfig config);
```

Comment:
i have added an encodingType (jwt/jsonld) as a mandatory attribute of the creation config. Since the ssi agent lib in the upstream does not support creating of VCs i'll add a switch, that'll throw NotImplementedYet when the type is JWT for VC

Reply:
I mean method `createCredential` should return `VerifiableCredential` and the call of this method should have respective tables(by creating `HoldersCredential` and `IssuerCredential`)

`createCredential` only signed the VC and return, 
In case of DB, it will get key from database and signed
In case of Azure, first it will create payload then send Azure to sign it

Reply:
ahh ok, that's waht you meant. We have 2 choices:
1. we just return VerifiableCredential
2. We return String (This way we can return the serialized credential, no matter if JWT or JsonLd - this is why createPresentation returns String)

Reply:
Option 2 is better I think